python/equistore-operations/equistore/operations/manipulate_labels_column.py

```diff
@@ -0,0 +1,310 @@
+    ...     tensor, index=0, name="bar", values=np.array([1]), axis="keys"
+    ... )
+    TensorMap with 1 blocks
+    keys: bar  foo
+           1    0
+    """
+    _check_axis(axis)
+
+    keys = tensor.keys
+    if axis == "keys":
+        keys = keys.insert(index=index, name=name, values=values)
+
+    blocks = []
+    for block in tensor:
+        samples = block.samples
+        properties = block.properties
+
+        if axis == "samples":
+            samples = samples.insert(index=index, name=name, values=values)
+        elif axis == "properties":
+            properties = properties.insert(index=index, name=name, values=values)
+
+        new_block = TensorBlock(
+            values=block.values,
+            samples=samples,
+            components=block.components,
+            properties=properties,
+        )
+
+        for parameter, gradient in block.gradients():
+            new_block.add_gradient(
+                parameter=parameter,
+                gradient=TensorBlock(
+                    values=gradient.values,
+                    samples=gradient.samples,
+                    components=gradient.components,
+                    properties=properties,
+                ),
+            )
+
+        blocks.append(new_block)
+
+    return TensorMap(keys=keys, blocks=blocks)
+
+
+def remove_labels_column(tensor: TensorMap, axis: str, name: str) -> TensorMap:
+    """Remove a :py:class:`equistore.Labels` column along the given axis.
+
+    Removal can only be performed if the resulting :py:class:`equistore.Labels`
+    instance will be unique.
+
+    For ``axis=="samples"`` the column is not removed from gradients.
+
+    :param tensor: the input :py:class:`TensorMap`.
+    :param axis: axis for which ``name`` should be removed. Allowed are ``"keys"``,
+                 ``"properties"`` or ``"samples"``.
+    :param name: the :py:class:`equistore.Labels` name to be removed
+
+    :raises ValueError: if ``axis`` is a not valid value
+    :raises ValueError: if name is not part of the axis
+
+    :return: a new :py:class:`equistore.TensorMap` with removed labels column.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import equistore
+    >>> values = np.array([[1, 2], [3, 4]])
+    >>> block = equistore.block_from_array(values)
+    >>> keys = equistore.Labels(["keys", "extra"], np.array([[0, 0]]))
+    >>> tensor = equistore.TensorMap(keys=keys, blocks=[block])
+    >>> tensor
+    TensorMap with 1 blocks
+    keys: keys  extra
+           0      0
+    >>> equistore.remove_labels_column(tensor, axis="keys", name="extra")
+    TensorMap with 1 blocks
+    keys: keys
+           0
+    """
+    _check_axis(axis)
+
+    keys = tensor.keys
+    if axis == "keys":
+        keys = keys.remove(name=name)
+
+    blocks = []
+    for block in tensor:
+        samples = block.samples
+        properties = block.properties
+
+        if axis == "samples":
+            samples = samples.remove(name)
```

Comment:
And maybe here too with the samples and properties
```suggestion
            samples = samples.remove(name)
            if len(samples) == 0:
                raise ValueError("Cannot remove the only samples name present.")
```